docs/design-docs/01-kpt-live-stdin.md

```diff
@@ -0,0 +1,242 @@
+apiVersion: kpt.dev/v1alpha1
+kind: ResourceGroup
+metadata  
+  name: <INVENTORY_NAME>
+  namespace: <INVENTORY_NAMESPACE>
+  labels:
+    cli-utils.sigs.k8s.io/inventory-id: <INVENTORY_ID>
+```
+
+### ResourceGroup as a singleton object
+
+`kpt live apply [--resourcegroup-path] -` from STDIN accepts and only accepts a single
+ResourceGroup, including the ResourceGroup provided by the flag. It detects
+1. If more than one ResourceGroup is found, raise errors and display all the ResourceGroup objects.
+2. If no ResourceGroup is found in STDIN and and Kptfile inventory does not exists, raise errors and suggest users to
+   run kpt live init
+3. If no ResourceGroup is found in STDIN and Kptfile inventory exists, raise errors and
+   suggest users to run kpt live migrate
+
+## User Guide
+
+### To hydrate via kustomize and deploy via kpt
+
+#### Day 1
+
+<b>For new users to start from scratch (no Kptfile)</b>
+User can run `kpt live init [--resourcegroup-file=CUSTOM_RG.yaml]` to create a
+ResourceGroup object and store it in a resourcegroup.yaml file.
+Users can customize the file path with the flag “--resourcegroup-file”.
+
+<b>For existing kpt users to migrate from Kptfile</b>
+Users run `kpt live migrate [--resourcegroup-file=CUSTOM_RG.yaml]` to convert the
+Inventory object from Kptfile to a standalone resourcegroup.yaml file.
+Users can customize the file path with the flag “--resourcegroup-file”.
+
+<b>[optional]: Add shareable ResourceGroup to kustomize resources</b>
+If the ResourceGroup is expected to be shared in the Gitops workflow, users can add
+the resourcegroup.yaml  file path to the .resources field in kustomization.yaml.
+This simplifies the Day N deployment by omitting the “–resourcegroup-file“ flag.
+
+#### Day N
+
+- Users can configure the hydration rules in kustomization.yaml
+- Users can run `kustomize build <DIR> | kpt live apply -` to hydrate and deploy
+  the kustomize-managed configurations. If resourcegroup.yaml is not added to
+  kustomize .resources field, users should provide the “–resourcegroup-file“ flag.  
+  `kustomize build <DIR> | kpt live apply –resourcegroup-file <CUSTOM_RG.yaml> -`
+
+### To hydrate via helm and deploy via kpt
+
+#### Day 1
+
+<b>For new users to start from scratch (no Kptfile)</b>
+User can run `kpt live init --resourcegroup-file=<DIR>/resourcegroup.yaml` to create
+a ResourceGroup object and store it in the helm template <DIR>.
+
+<b>For existing kpt users to migrate from Kptfile</b>
+Users run `kpt live migrate --resourcegroup-file=<DIR>/resourcegroup.yaml` to convert
+the Inventory object from Kptfile to  a standalone resourcegroup.yaml file.
+
+#### Day N
+
+Users can run `helm templates <DIR> | kpt live apply -` to hydrate and deploy the
+helm-managed resources.
+
+See kpt issue #2399 for expected Inventory usage in package publisher/consumer.
+
+## FAQ
+
+### Will Inventory be deprecated from Kptfile?
+
+Kpt still supports inventory in Kptfile and it is not required to migrate to the
```

Comment:
I would consider deprecating it in the future and talking about the timeline

Reply:
yeah, Sunil Mortan and I have discussed about it. And we want to have more signals before moving forward. Deprecating the inventory might be aggressive without user data support.